src/utils/wallet/bitget_wallet.ts

```
@@ -0,0 +1,174 @@
+import { WalletProvider, Network, Fees, UTXO } from "./wallet_provider";
+import {
+  getAddressBalance,
+  getTipHeight,
+  getFundingUTXOs,
+  getNetworkFees,
+  pushTx,
+} from "../mempool_api";
+import { Psbt } from 'bitcoinjs-lib'
+
+type BitgetWalletInfo = {
+  publicKeyHex: string;
+  address: string;
+};
+
+export class BitgetWallet extends WalletProvider {
+  private bitgetWalletInfo: BitgetWalletInfo | undefined;
+  private provider = window?.bitkeep?.unisat
+  constructor() {
+    super();
+  }
+
+  connectWallet = async (): Promise<any> => {
+    // check whether there is an Bitget Wallet extension
+    if (!this.provider) {
+      throw new Error("Bitget Wallet extension not found");
+    }
+
+    try {
+      await this.provider?.requestAccounts() // Connect to Bitget Wallet extension
+      await this.provider?.switchNetwork('signet')
+    } catch (error) {
+      if ((error as Error)?.message?.includes('rejected')) {
+        throw new Error('Connection to Bitget Wallet was rejected')
+      } else {
+        throw new Error((error as Error)?.message)
+      }
+    }
+
+    const address = await this.getAddress()
+    const publicKeyHex = await this.getPublicKeyHex()
+
+    if (address && publicKeyHex) {
+      this.bitgetWalletInfo = {
+        publicKeyHex,
+        address,
+      };
+      return this;
+    } else {
+      throw new Error("Could not connect to Bitget Wallet");
+    }
+  };
+
+  getWalletProviderName = async (): Promise<string> => {
+    return 'Bitget Wallet';
+  };
+
+  getAddress = async (): Promise<string> => {
+    let accounts = (await this.provider?.getAccounts()) || []
+    return accounts[0]
+  };
+
+  getPublicKeyHex = async (): Promise<string> => {
+    let publicKey = await this.provider?.getPublicKey()
+    if (!publicKey) {
+      console.log('bitget Wallet not connected')
```

We should throw an error here

---

Done